src/main/java/pl/debuguj/parkingspacessystem/services/ParkingSpaceManagementService.java

```diff
@@ -0,0 +1,44 @@
+package pl.debuguj.parkingspacessystem.services;
+
+import pl.debuguj.parkingspacessystem.exceptions.CarRegisteredInSystemException;
+import pl.debuguj.parkingspacessystem.exceptions.IncorrectEndDateException;
+import pl.debuguj.parkingspacessystem.domain.ParkingSpace;
+import pl.debuguj.parkingspacessystem.exceptions.ParkingSpaceNotFoundException;
+
+import java.math.BigDecimal;
+import java.util.Date;
+
+/**
+ * Created by grzesiek on 09.10.17.
+ */
+public interface ParkingSpaceManagementService {
+
+    /**
+     As a driver, I want to start the parking meter, so I don’t have to pay
+     the fine for the invalid parking
+     */
+    /**
+     As a driver, I want to know how much I have to pay for parking
+     */
+    BigDecimal reserveParkingSpace(final ParkingSpace ps) throws CarRegisteredInSystemException;
```

Comment:
jaki to ma sens? Parkujący rezerwuje i musi od razu podać czas końca? Nie sądzisz, że łatwo będzie shackować taki system? ;) Jak potem parkujący ma sprawdzać czas swojego parkowania? Wprowadzając nowe parkowanie?

Reply:
Założyłem najprostszy wariant, czyli, że kierowca od razu podaje czas parkowania. Nie chciałem się wdawać w wiekszą implementację. Masz rację, że w praktyce to bez sensu.